### ISSUE TITLE
illegal instruction on start - even with native compiling

### ISSUE THREAD

**Issue by fischer-hub** [bbuchfink/diamond#540]
Hey, such a nice project! 
Although I'm having quite some problems getting `diamond` (v2.0.13) to run on our HPC, the prebuilt binaries (as well as the conda installation) all fail to start at all, throwing 'illegal instruction' errors.

The HPC is running Debian GNU/Linux 11 (bullseye) on arch:

```Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   40 bits physical, 48 bits virtual
CPU(s):                          24
On-line CPU(s) list:             0-23
Thread(s) per core:              2
Core(s) per socket:              6
Socket(s):                       2
NUMA node(s):                    2
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           44
Model name:                      Intel(R) Xeon(R) CPU           E5649  @ 2.53GHz
Stepping:                        2
CPU MHz:                         1596.000
BogoMIPS:                        5053.98
L1d cache:                       384 KiB
L1i cache:                       384 KiB
L2 cache:                        3 MiB
L3 cache:                        24 MiB
NUMA node0 CPU(s):               0,2,4,6,8,10,12,14,16,18,20,22
NUMA node1 CPU(s):               1,3,5,7,9,11,13,15,17,19,21,23
Vulnerability Itlb multihit:     KVM: Mitigation: VMX unsupported
Vulnerability L1tf:              Mitigation; PTE Inversion
Vulnerability Mds:               Vulnerable: Clear CPU buffers attempted, no microcode; SMT vulnerable
Vulnerability Meltdown:          Mitigation; PTI
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Full generic retpoline, IBPB conditional, IBRS_FW, STIBP conditional, RSB fill
                                 ing
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi m
                                 mx fxsr sse sse2 ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts
                                  rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl 
                                 smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 popcnt aes lahf_lm pti ssbd ibrs i
                                 bpb stibp dtherm ida arat flush_l1d
```

Sadly I can't give any `diamond` log files, since `diamond` doesn't really start at all.
I already tried to build from source like described in the [wiki](https://github.com/bbuchfink/diamond/wiki/2.-Installation#compiling-from-source) with 'cmake -DCMAKE_BUILD_MARCH=native' set, but I still get the same error on start.. I'm not quite sure whats going wrong here, any ideas?

gdb output:
```
Reading symbols from diamond...
(No debugging symbols found in diamond)
(gdb) run
Starting program: /home/fisched99/humann/diamond-2.0.13/bin/diamond version
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGILL, Illegal instruction.
0x00005555556d26c1 in _GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv ()
(gdb) bt
#0  0x00005555556d26c1 in _GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv ()
#1  0x0000555555d9ea05 in __libc_csu_init ()
#2  0x00007ffff7c77c9a in __libc_start_main (main=0x5555556bd0d0 <main>, argc=2, argv=0x7fffffffdad8, 
    init=0x555555d9e9c0 <__libc_csu_init>, fini=<optimized out>, rtld_fini=<optimized out>, 
    stack_end=0x7fffffffdac8) at ../csu/libc-start.c:264
#3  0x00005555556d2faa in _start ()

(gdb) layout asm
│  >0x5555556d26c1 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+49>         vmovdqa 0x6fc8b7(%rip),%ymm0        # 0x555555dcef80                                            │
│   0x5555556d26c9 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+57>         lea    0x6000(%rax),%rdx                                                                        │
│   0x5555556d26d0 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+64>         vmovdqa %ymm0,(%rax)                                                                            │
│   0x5555556d26d4 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+68>         add    $0x60,%rax                                                                               │
│   0x5555556d26d8 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+72>         vmovdqa %ymm0,-0x40(%rax)                                                                       │
│   0x5555556d26dd <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+77>         vmovdqa %ymm0,-0x20(%rax)                                                                       │
│   0x5555556d26e2 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+82>         cmp    %rax,%rdx                                                                                │
│   0x5555556d26e5 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+85>         jne    0x5555556d26d0 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+64>          │
│   0x5555556d26e7 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+87>         lea    0x884eb2(%rip),%rax        # 0x555555f575a0 <_ZN9Benchmark9ARCH_AVX2L14horizontal_gapE>  │
│   0x5555556d26ee <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+94>         lea    0x6000(%rax),%rdx                                                                        │
│   0x5555556d26f5 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+101>        nopl   (%rax)                                                                                   │
│   0x5555556d26f8 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+104>        vmovdqa %ymm0,(%rax)                                                                            │
│   0x5555556d26fc <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+108>        add    $0x60,%rax                                                                               │
│   0x5555556d2700 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+112>        vmovdqa %ymm0,-0x40(%rax)                                                                       │
│   0x5555556d2705 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+117>        vmovdqa %ymm0,-0x20(%rax)                                                                       │
│   0x5555556d270a <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+122>        cmp    %rdx,%rax                                                                                │
│   0x5555556d270d <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+125>        jne    0x5555556d26f8 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+104>         │
│   0x5555556d270f <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+127>        mov    $0x400,%edx                                                                              │
│   0x5555556d2714 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+132>        mov    $0x80,%esi                                                                               │
│   0x5555556d2719 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+137>        lea    0x884980(%rip),%rdi        # 0x555555f570a0 <_ZN9Benchmark9ARCH_AVX2L7profileE>          │
│   0x5555556d2720 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+144>        vzeroupper                                                                                      │
│   0x5555556d2723 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+147>        pop    %rbp                                                                                     │
│   0x5555556d2724 <_GLOBAL__sub_I__ZN9Benchmark9ARCH_AVX217swipe_cell_updateEv+148>        jmp    0x555555679110 <memset@plt>                                                              │
│   0x5555556d2729                                                                              nopl   0x0(%rax)                                                                            │
│   0x5555556d2730 <_GLOBAL__sub_I__ZN2DP11BandedSwipe9ARCH_AVX23binE9HspValuesiiilji>      sub    $0x8,%rsp                                                                                │
│   0x5555556d2734 <_GLOBAL__sub_I__ZN2DP11BandedSwipe9ARCH_AVX23binE9HspValuesiiilji+4>    call   0x55555576e480 <_ZN4SIMD4archEv>                                                         │
```

Happy holidays!

**Comment by fischer-hub** [comment]
Hmm I just tested with `diamond` v2.0.9 and its running like a charm.. well that will be sufficient to me, but it seems like theres a bug in the newest release that causes the binarys to always contain AVX instructions, even when compiling with march=native and -no-avx on a non AVX cpu..

**Comment by bbuchfink** [comment]
This should fix it: e43202ebb8b7b880840635def29df2698c409a9d

**Comment by fischer-hub** [comment]
Yes its fixed now! Thanks a lot!
